cdap-ui/app/cdap/components/Number/index.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React from 'react';
+import ThemeWrapper from 'components/ThemeWrapper';
+import PropTypes from 'prop-types';
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import TextField from '@material-ui/core/TextField';
+
+const styles = (theme) => {
+  return {
+    input: {
+      padding: 10,
+    },
+    root: {
+      // this doesn't seem to change hover styling
+      // '&:hover $notchedOutline': {
+      //   borderColor: '#cccccc',
+      // },
+      '&$focused $notchedOutline': {
+        border: '1px solid #098cf9',
+      },
+    },
+    focused: {},
+    notchedOutline: {},
+  };
+};
+
+interface INumberProps extends WithStyles<typeof styles> {
+  value: string;
+  onChange: (value: string) => void;
+  disabled: boolean;
+  isFieldRequired: boolean;
+}
+
+function Number({ value, onChange, disabled, isFieldRequired, classes }: INumberProps) {
+  // not required and not disabled by default
+  // default value for number input?
```

Comment:
is this comment required? We can add comments if its helpful for future reference. Otherwise we can skip it.

Reply:
I have deleted all comments and will be more deliberate about them in the future!